transport/http2_server.go

```diff
@@ -274,8 +274,11 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 func (t *http2Server) HandleStreams(handle func(*Stream), traceCtx func(context.Context, string) context.Context) {
 	// Check the validity of client preface.
 	preface := make([]byte, len(clientPreface))
-	if _, err := io.ReadFull(t.conn, preface); err != nil {
-		grpclog.Printf("transport: http2Server.HandleStreams failed to receive the preface from client: %v", err)
+	if n, err := io.ReadFull(t.conn, preface); err != nil {
+		// Only log if it isn't a simple tcp accept check (ie: tcp balancer doing open/close socket)
+		if n > 0 || err != io.EOF {
```

Comment:
I believe we just need to check `io.EOF` here. If n>0, err will be `ErrUnexpectedEOF`.

From the doc of `ReadFull`:

    The error is EOF only if no bytes were read.
    If an EOF happens after reading some but not all the bytes,
    ReadFull returns ErrUnexpectedEOF.